modules/exoplanets/datastructure.cpp

```diff
@@ -0,0 +1,332 @@
+/*****************************************************************************************
+ *                                                                                       *
+ * OpenSpace                                                                             *
+ *                                                                                       *
+ * Copyright (c) 2014-2024                                                               *
+ *                                                                                       *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of this  *
+ * software and associated documentation files (the "Software"), to deal in the Software *
+ * without restriction, including without limitation the rights to use, copy, modify,    *
+ * merge, publish, distribute, sublicense, and/or sell copies of the Software, and to    *
+ * permit persons to whom the Software is furnished to do so, subject to the following   *
+ * conditions:                                                                           *
+ *                                                                                       *
+ * The above copyright notice and this permission notice shall be included in all copies *
+ * or substantial portions of the Software.                                              *
+ *                                                                                       *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,   *
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A         *
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT    *
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF  *
+ * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE  *
+ * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                                         *
+ ****************************************************************************************/
+
+#include <modules/exoplanets/datastructure.h>
+
+#include <modules/exoplanets/exoplanetshelper.h>
+#include <openspace/scene/scene.h>
+#include <openspace/util/distanceconstants.h>
+#include <openspace/util/time.h>
+#include <openspace/util/timeconversion.h>
+#include <ghoul/logging/logmanager.h>
+#include <string_view>
+
+namespace {
+    constexpr std::string_view _loggerCat = "ExoplanetsModule";
+
+    // KeplerTranslation requires angles in range [0, 360]
+    float validAngle(float angle, float defaultValue, bool& usedDefault) {
```

Comment:
Hm, I get your point. However, since I also want to keep track of whether the default value was used, that would bloat the code in the function below quite a lot.

Since it's a very specific function for this code only, maybe I should just put it in the `toDataDictionary` function instead? I used to have it as a lambda function that captured the `hasUsedDefaultValues` variable and just altered that variable and maybe that was better